claassenlab/Teamproject-2023
# Feature #10, #41: Created initial analysis menu and integrated UMAP projections

## Motivation

Resolves #10
Resolves #41

## Changes


It is now possible to open an analysis menu window by clicking the `Analysis menu` button. There is already one section for UMAP projections including an OFF/ON switch as well as a dropdown menu with four possible colors to choose from. By default, the color `default` is enabled and it is only possible to enable one color at the same time. If there is a loom file loaded and the UMAP check box is enabled, it is possible to run the analysis by clicking the blue `Run analysis` button. Then, scanpy opens a window with the respective UMAP projection. **But please note that the data used for these visualizations has nothing to do with the data from the loaded loom file because I haven't found a scRNA loom file to use for testing. If there is one in the future, one just need to change some few lines of code I've already marked.**

## Testing


- Load the loom example file (also possible after analysis menu)
- Open the analysis menu and enable UMAP
- Maybe change the color (try every color)
- Click `Run analysis` and see what happens

## Additional context


This is only the base for the future as the analysis menu is very much unfinished at this state and there are many more options for UMAP and all the other analysis things we want to implement in the future. Nevertheless, it just is our first visualization ever and it seems to work quite well. Thank you `scanpy`.
